File: creational/singleton/typescript/src/realLife/clientCode.ts
import { EventService } from "./EventService";

const eventService = EventService.instance;

const eventSubscription = eventService.subscribe((event) => console.log(event));

eventService.emit("First Event");

EventService.instance.emit("Second Event");

eventSubscription.unsubscribe();

EventService.instance.emit("Third event"); // Will not emit

File: #3.design-patterns/creational/singleton/dart/realLife/clientCode.dart
import 'EventService.dart';

void main() {
  var eventService = EventService();

  var eventSubscription = eventService.subscribe((value) {
    print(value);
  });

  var eventSubscription2 = eventService.subscribe((value) {
    print("${value} eventSubscription2");
  });

  eventService.emit("First event");
  eventService.emit("Second event");

  eventSubscription.unsubscribe();

  eventService.emit("Third event");
  eventService.emit("Fourth event");

  eventSubscription2.unsubscribe();
}
